build.gradle

```diff
@@ -46,6 +56,7 @@ buildscript {
 plugins {
     id 'io.spring.dependency-management' version '1.0.10.RELEASE'
     id 'com.diffplug.spotless' version '5.6.1'
+    id 'org.openapi.generator' version '4.3.1'
```

Comment:
Hmm... I thought we wanted to only have "common" plugins in the main Gradle file. But the OpenAPI one is only needed / used on the client project. Is there any way to make it specific for that subproject only?

Reply:
This is just to define the plugin dependencies. We could also do something like this:

```
...
id 'org.openapi.generator' version '4.3.1' apply false
...
```

Then this is really just declaring a dependency (much like the dependencyManagement section) and the respective plugins would only be activated with (same example):

```
apply plugin: 'org.openapi.generator'
```

Reply:
On that note: all those plugin dependencies declared like this in the buildscript section:

```
    dependencies {
        classpath 'com.bmuschko:gradle-cargo-plugin:2.7.1'
        classpath 'org.zeroturnaround:gradle-jrebel-plugin:1.1.10'
        classpath 'org.springframework.boot:spring-boot-gradle-plugin:2.3.4.RELEASE'
        classpath 'gradle.plugin.com.hierynomus.gradle.plugins:license-gradle-plugin:0.15.0'
        classpath 'org.apache.openjpa:openjpa:3.1.2' // when upgrading, also change OpenJPA version repeated in fineract-provider/build.gradle!
        classpath 'com.radcortez.gradle:openjpa-gradle-plugin:3.1.0'
        classpath 'org.nosphere.apache:creadur-rat-gradle:0.7.0'
        classpath "gradle.plugin.com.github.spotbugs.snom:spotbugs-gradle-plugin:4.5.1"
        classpath "gradle.plugin.com.github.andygoossens:gradle-modernizer-plugin:1.3.0"
        classpath "gradle.plugin.com.gorylenko.gradle-git-properties:gradle-git-properties:2.2.3"
        classpath "net.ltgt.gradle:gradle-errorprone-plugin:1.2.1"
        classpath "io.swagger.core.v3:swagger-gradle-plugin:2.1.5"
        classpath "gradle.plugin.org.hidetake:gradle-swagger-generator-plugin:2.18.2"
    }
```

... are considered deprecated Gradle style (will probably not be supported in Gradle 7).

Reply:
I've added "apply false" in the plugin section; that way we are sure that those plugins are not activated globally (with potential side effects). FYI